### PR TITLE
[processing] Don't show hidden parameters in processing.algorithmHelp() output

### PR DESCRIPTION
**Pull request by agiudiceandrea**
## Description

Hidden parameters (defined with flag `QgsProcessingParameterDefinition.FlagHidden`) of processing algorithms are incorrectly displayed in the `processing.algorithmHelp()` output.

This PR avoids that behaviour.

### CODE REVIEW

File: python/plugins/processing/tools/general.py [hunk 1]
```diff
@@ -53,11 +54,13 @@ def algorithmHelp(id):
...
         print('----------------')
         for p in alg.parameterDefinitions():
+            if p.flags() & QgsProcessingParameterDefinition.FlagHidden:
+                continue
             print(f'\n{p.name()}: {p.description()}')
             if p.help():
                 print(f'\n\t{p.help()}')
 
-            print(f'\n\tParameter type:\t{p.__class__.__name__}')
+            print(f'\n\tParameter type: {p.__class__.__name__}')
```
Comment (nyalldawson) [inline python/plugins/processing/tools/general.py:63]: why this change?

Comment (agiudiceandrea) [reply python/plugins/processing/tools/general.py:63]: It looks like such tabulation character is badly displayed in the Python console on Windows
image
and I think it's useless and in other strings the colon is always followed by a space.
If OK, I would also remove the double space in 
https://github.com/qgis/QGIS/blob/11f083b6c0eb67c8d119c9c8a631ccac5d2489e5/python/plugins/processing/tools/general.py#L81-L82
Otherwise I can restore the tabulation.

Comment (nyalldawson) [reply python/plugins/processing/tools/general.py:63]: Hm, odd -- because the lines just above that use tab characters!

My gut feeling is that this is just because we (incorrectly) try to parse any outputs in the Python console as python code itself. I'd say we leave the current behavior and fix the issue in the console instead...

Comment (agiudiceandrea) [reply python/plugins/processing/tools/general.py]: @nyalldawson, OK. Tab restored and extra space not removed.

**Review by nyalldawson — APPROVED** [review]
